### ISSUE TITLE
yabai-m display --focus sometimes mysteriously fails

### ISSUE THREAD

**Issue by alextremblay** [koekeishiya/yabai#684]
Hello again,

Thanks once again for making such a fantastic tool.

I've encountered a rather curious bug when trying to change focus from one display to another. 
I have 3 displays. 

If i run command `yabai -m display --focus 2`, focus jumps instead to display index 1. this happens regardless of which dispaly I'm on when I run the command.
If I run command `yabai -m window --display 2 && yabai -m display --focus 2` (ie "move window to display and follow), the window moves to display 2, but focus still jumps to display 1

None of this happens if i try to focus or move a window from any of the displays to any other display, only happens when moving to / focussing display 2

Stranger still, this only happens if that display is on a particular space. If I switch to a different space on that display, the problem goes away and everything works as expected.

I'm not sure what kind of information you need to troubleshoot this, but here's everything I can gather right now.

`yabai -m query --displays`:
```json
[{
        "id":69975444,
        "uuid":"E7522739-5F2D-3324-7F3B-1D331987BAAF",
        "index":1,
        "spaces":[1, 2, 3, 4, 5, 6, 7, 8],
        "frame":{
                "x":0.0000,
                "y":0.0000,
                "w":2048.0000,
                "h":1152.0000
        }
},{
        "id":722478418,
        "uuid":"6AA1F64E-7740-27B7-24C8-B79F6DA2E03F",
        "index":3,
        "spaces":[12, 13],
        "frame":{
                "x":-1920.0000,
                "y":36.0000,
                "w":1920.0000,
                "h":1080.0000
        }
},{
        "id":722489923,
        "uuid":"80FC8E10-0500-33C8-163C-6BB492C35848",
        "index":2,
        "spaces":[9, 10, 11],
        "frame":{
                "x":2048.0000,
                "y":-382.0000,
                "w":1080.0000,
                "h":1920.0000
        }
}]
```

space number 9 is the one where it's not working.
`yabai -m query --spaces --space 9`:
```json
{
        "id":16,
        "label":"",
        "index":9,
        "display":2,
        "windows":[136151, 136150, 62193, 73700, 126633],
        "type":"bsp",
        "visible":1,
        "focused":0,
        "native-fullscreen":0,
        "first-window":62193,
        "last-window":73700
}
```

and when display 2 is on space 10, everything works
`yabai -m query --spaces --space 10`:
```json
{
        "id":14,
        "label":"",
        "index":10,
        "display":2,
        "windows":[75069],
        "type":"bsp",
        "visible":1,
        "focused":0,
        "native-fullscreen":0,
        "first-window":75069,
        "last-window":75069
}
```

Any thoughts?

Any other info you'd like me to grab?

**Comment by koekeishiya** [comment]
This sounds like some bizarre issue caused by some macOS cache or something. My suggestion would be to destroy all spaces on display 2 and re-create them one by one.

**Comment by alextremblay** [comment]
Hi koekeishya,

apologies for not getting back to you regarding this, i completely missed the notification regarding your suggestion. 

I did as you suggested, and that did indeed fix the problem... for a while, and then it came back.

I've since switched over to Amethyst (sorry), and I still occasionally experience the same issue, but not as often. I suspect this has something to do with the fact that two of my 3 displays are displaylink displays, not real displays (I'm on am M1 macbook, which only supports one external display). I should have mentioned this at the time, but i hadn't even considered it as a potential culprit.

I think this is a displaylink+macos bug, far beyond yabai's ability to fix, you were right to close this ticket.

Cheers
